types/logger/logger_test.go

```diff
@@ -19,3 +20,19 @@ func TestStdLogger(t *testing.T) {
 	lg := StdLogger(t.Logf)
 	lg.Printf("plumbed through")
 }
+
+func TestRateLimiter(t *testing.T) {
+	lg := RateLimitedFn(t.Logf, 1, 1)
```

Comment:
I don't think you actually want to use t.Logf() here. Instead, you want to write your own fake logger function that validates the input it was passed at different times. (The easiest way to do this is to define a logf local function that has access to variables from the test function, such as 't' and some variable that reflects whether you're expecting a message to be printed or not.)

This test doesn't call any function like t.Errorf() or t.Fatalf() here, so right now it can never fail. Your fake logf() function could call t.Fatalf() if the rate limiter tries to print at the "wrong" time.